src/createRollupFetchTransactionHash.ts

```diff
@@ -0,0 +1,47 @@
+import { deterministicFactoriesDeploymentEstimatedFees } from './constants';
+import { Address, PublicClient } from 'viem';
+import { AbiEvent } from 'abitype'
+
+const RollupInitializedEventAbi: AbiEvent = {
+  anonymous: false,
+  inputs: [
+      {
+          indexed: false,
+          internalType: "bytes32",
+          name: "machineHash",
+          type: "bytes32",
+      },
+      {
+          indexed: false,
+          internalType: "uint256",
+          name: "chainId",
+          type: "uint256",
+      },
+  ],
+  name: "RollupInitialized",
+  type: "event",
+};
+
+export async function createRollupFetchTransactionHash(
+  rollupAddress: Address,
+  publicClient: PublicClient,
+) {
```

Comment:
I suggest we always pass objects to these functions, so we have nice parameter naming when we call them

```suggestion
export async function createRollupFetchTransactionHash({
  rollup,
  publicClient
} : {
  rollup: Address,
  publicClient: PublicClient,
}) {
```